ambry-commons/src/test/java/com.github.ambry.commons/MockHelixPropertyStore.java

```diff
@@ -102,9 +102,19 @@ public boolean set(String path, T record, int expectVersion, int options) {
 
   @Override
   public boolean update(String path, DataUpdater<T> updater, int options) {
-    T newRecord = updater.update(pathToRecords.get(path));
-    System.out.println("Updating to store path: " + path + ", record: " + newRecord.toString());
-    return setAndNotify(path, newRecord);
+    T newRecord = null;
+    boolean exceptionDuringUpdater = false;
+    try {
+      newRecord = updater.update(pathToRecords.get(path));
+    } catch (Exception e) {
+      exceptionDuringUpdater = true;
```

Comment:
As discussed offline, we want this method to fully mimic the behavior of the actual implementation, which will help us to test the failure path, i.e., we hope to test when returning `false`.